doc/src/programmers-guide.md

```diff
@@ -1262,7 +1262,7 @@ The AtomVM virtual machine and libraries support APIs for interfacing with perip
 
 The GPIO peripheral has nif support on all platforms. One notable difference on the STM32 platform is that `Pin()` is defined as a tuple consisting of the bank (a.k.a. port) and pin number.  For example a pin labeled PB7 on your board would be `{b,7}`.
 
-You can read and write digital values on GPIO pins using the [`gpio` module](./apidocs/erlang/eavmlib/gpio.md), using the [`digital_read/1`](./apidocs/erlang/eavmlib/gpio.md#digital_read1) and [`digital_write/2`](./apidocs/erlang/eavmlib/gpio.md#digital_write2) functions.  You must first set the direction of the pin using the [`gpio:set_direction/2`](./apidocs/erlang/eavmlib/gpio.md#set_direction3) function, using `input` or `output` as the direction parameter.
+You can read and write digital values on GPIO pins using the [`gpio` module](./apidocs/erlang/eavmlib/gpio.md), using the [`digital_read/1`](./apidocs/erlang/eavmlib/gpio.md#digital_read1) and [`digital_write/2`](./apidocs/erlang/eavmlib/gpio.md#digital_write2) functions.  You must first set the direction of the pin using the [`gpio:set_direction/3`](./apidocs/erlang/eavmlib/gpio.md#set_direction3) function, using `input` or `output` as the direction parameter.
```

Comment:
`gpio:set_direction/3` is for the port driver, the nif uses `gpio:set_direction/2`. Any of the functions that take a GPIO parameter are strictly for use with the port driver, those without a GPIO parameter are all nifs.